File: src/main/java/ast/ASTNode.java
package ast;

/* all ast nodes inherit from this */
public class ASTNode {
    /* start location in file */
    ASTFileLocation loc;

    public ASTNode(ASTFileLocation loc){
        this.loc = loc;
    }

    void printIndent(int i){
        for(var n = 0; n < i; n++){
            System.out.print("\t");
        }
    }

    public void print(int i){
        printIndent(i);
        System.out.println("ASTNode");
    }

}

File: src/main/java/ast/ASTNode.kt
package ast

/* all ast nodes inherit from this */
open class ASTNode(/* start location in file */
        internal var loc: ASTFileLocation?) {

     open fun printIndent(i: Int) {
        for (n in 0 until i) {
            System.out.print("\t")
        }
    }

    open fun print(i: Int) {
        printIndent(i)
        System.out.println("ASTNode")
    }

}